rust/hyperlane-core/src/rpc_clients/fallback.rs

```diff
@@ -130,25 +137,66 @@ where
                 .await;
         }
     }
+
+    /// Call the first provider, then the second, and so on (in order of priority) until a response is received.
+    /// If all providers fail, return an error.
+    pub async fn call<V>(
```

Comment:
noting for posterity that we chatted about this having a lot of overlap with https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/e6d060fd371216c5ecdddbb46cedc9dc48f49c89/rust/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs#L86 - moving the ethereum fallback provider to using this is possible but to avoid spending too much time we'll just keep as is

Reply:
updated that comment in `hyperlane-ethereum` 🫡